cranelift/codegen/src/machinst/abi.rs

```diff
@@ -472,6 +475,38 @@ pub trait ABIMachineSpec {
     /// Generate a meta-instruction that adjusts the nominal SP offset.
     fn gen_nominal_sp_adj(amount: i32) -> Self::I;
 
+    /// When setting up for a call, ensure that `space` bytes are available in the outgoing
+    /// argument area on the stack for arguments to that function, and any values returned through
+    /// the stack.
```

Comment:
```suggestion
    /// When setting up for a call, ensure that `space` bytes are available in the outgoing
    /// argument area on the stack. The specified amount of space is the minimum required for both
    /// arguments to that function, and any values returned through
    /// the stack. There are two reasonable implementations which each target can choose between:
    /// 1. At least this much space is reserved during the prologue and `StackAMode::OutgoingArg` refers
    ///    to the bottom of the reserved area, so this method does nothing.
    /// 2. `StackAMode::OutgoingArg` refers to the top of this area, so this method needs to adjust the stack
    ///    pointer to trim any unused portion of the bottom of the stack frame immediately before the call.
    /// `gen_restore_argument_area` needs to undo any stack pointer changes made here.
```

(Sorry, I can't easily word-wrap this in the web interface.)